### PR TITLE
Universal link support

### LINKED ISSUE #35: Add Universal Link support
This is dependent on https://github.com/5calls/5calls/issues/75 to request issue details for a specific issue id.

### PR DESCRIPTION
**Pull request by nickoneill**
Fixes #35 

Support on the website for this is already complete so you can build this, go to an issue link at 5calls.org and see the automatic "Open in the 5 Calls app" link at the top of Safari. I opted not to make it automatic for now.

Todo:
- [x] Track an opening from a universal link in our event tracking

### CODE REVIEW

File: FiveCalls/FiveCalls/IssuesViewController.swift [hunk 1]
```diff
@@ -155,12 +156,33 @@ class IssuesViewController: UIViewController, UITableViewDelegate, UITableViewDa
...
+            // set selected issue
+            let selectedIssue = issuesManager.issues.first(where: {$0.slug == selectPath})
+            autoSelectIssue = selectedIssue
+            
+            if let split = self.splitViewController {
+                // split contexts are shown manually
+                showiPadIssueDetail(split: split)
+            } else {
+                // non-split contexts use segues
+                self.performSegue(withIdentifier: R.segue.issuesViewController.issueSegue.identifier, sender: nil)
```
Comment (mccarron) [inline FiveCalls/FiveCalls/IssuesViewController.swift:180]: Don't need self here?

File: FiveCalls/FiveCalls/IssuesViewController.swift [hunk 2]
```diff
@@ -187,32 +209,51 @@ class IssuesViewController: UIViewController, UITableViewDelegate, UITableViewDa
...
+        // don't need to unset this after use because it's always a fallback to the table selection
+        if let issue = autoSelectIssue {
+            return issue
+        }
+        
+        return nil
+    }
+
+    func showiPadIssueDetail(split: UISplitViewController) {
+        guard let selectedIssue = self.selectedIssue() else { return }
```
Comment (mccarron) [inline FiveCalls/FiveCalls/IssuesViewController.swift:234]: More inconsistent usage of `self`

File: FiveCalls/FiveCalls/IssuesViewController.swift [hunk 3]
```diff
@@ -155,12 +156,33 @@ class IssuesViewController: UIViewController, UITableViewDelegate, UITableViewDa
...
             tableView.reloadEmptyDataSet()
         }
         tableView.refreshControl?.endRefreshing()
+
+    }
+
+    func moveForwardIfNeeded() {
+        if let selectPath = UserDefaults.standard.string(forKey: UserDefaultsKey.selectIssuePath.rawValue) {
+            // set selected issue
+            let selectedIssue = issuesManager.issues.first(where: {$0.slug == selectPath})
```
Comment (mccarron) [inline FiveCalls/FiveCalls/IssuesViewController.swift:172]: Curious if this is worth adding to `issuesManager` to make it easy to look up an issue from a slug from anywhere else? I can see that being useful.

**Comment by nickoneill** [comment]
Thanks @mccarron! I will make the style adjustments when I add tracking later today

**Comment by subdigital** [comment]
Thanks @mccarron, agree with your comments.

File: FiveCalls/FiveCalls/IssuesViewController.swift (hunk 1 above)
Comment (nickoneill) [reply FiveCalls/FiveCalls/IssuesViewController.swift]: Fixed

File: FiveCalls/FiveCalls/IssuesViewController.swift (hunk 2 above)
Comment (nickoneill) [reply FiveCalls/FiveCalls/IssuesViewController.swift]: Fixed

File: FiveCalls/FiveCalls/IssuesViewController.swift (hunk 3 above)
Comment (nickoneill) [reply FiveCalls/FiveCalls/IssuesViewController.swift]: Added this to issuesManager instead

**Pull request merged**